### PR TITLE
Add MEGAN/DAA2INFO

### LINKED ISSUE #1847: new module: MEGAN/DAA2INFO
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #1847 

Basically a clone of MEGAN/RMA2INFO but with daa files as input

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/megan/daa2info/meta.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: "megan_daa2info"
+description: Analyses an DAA file and exports information in text format
```
Comment (JoseEspinosa) [inline modules/megan/daa2info/meta.yml:2]:
```suggestion
description: Analyses a DAA file and exports information in text format
```

File: modules/megan/daa2info/meta.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - daa:
+      type: file
+      description: DAA file from DIAMOND
+      pattern: "*.daa"
+  - megan_summary:
+      type: boolean
+      description: Specify whether to generate an MEGAN summary file
```
Comment (JoseEspinosa) [inline modules/megan/daa2info/meta.yml:30]:
```suggestion
      description: Specify whether to generate a MEGAN summary file
```

File: modules/megan/daa2info/main.nf [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/megan:6.21.7--h9ee0642_0':
+        'quay.io/biocontainers/megan:6.21.7--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(daa)
+    val(megan_summary)
+
+    output:
+    tuple val(meta), path("*.txt.gz")               , emit: txt
```
Comment (JoseEspinosa) [inline modules/megan/daa2info/main.nf:15]:
```suggestion
    tuple val(meta), path("*.txt.gz")               , emit: txt_gz
```

File: modules/megan/daa2info/main.nf [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    tuple val(meta), path("*.megan"), optional: true, emit: megan_summary
```
Comment (JoseEspinosa) [inline modules/megan/daa2info/main.nf:16]: The convention is not to name them as the file is named?
```suggestion
    tuple val(meta), path("*.megan"), optional: true, emit: megan
```

File: modules/megan/daa2info/meta.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - txt:
```
Comment (JoseEspinosa) [inline modules/megan/daa2info/meta.yml:42]:
```suggestion
  - txt_gz:
```

**Review by JoseEspinosa — APPROVED** [review]
LGTM! 🚀
Just made some small suggestions, feel free to merge once you accept/reject them 😄

File: modules/megan/daa2info/meta.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      type: file
+      description: Compressed text file
+      pattern: "*.txt.gz"
+  - megan_summary:
```
Comment (jfy133) [inline modules/megan/daa2info/meta.yml:46]:
```suggestion
  - megan:
```

**Pull request merged**